website/docs/reference/web/portals-plugin.md

```diff
@@ -74,18 +74,17 @@ type SubscriptionCallback<T = unknown> = (result: { topic: string, data: T; }) =
 
 ### publish
 
-Publishes some data to a provided topic.
+Publishes some data to a provided topic. Type-safety supported through the optional `TMessage` generic type.
 
 #### Usage 
 
 ```typescript
-const message: PortalMessage = {
-    topic: 'dismiss',
-    data: 'cancel',
-}
+type Messages = 
+  | { topic: 'modal:dismiss', data: 'cancel' | 'failure' }
+  | { topic: 'modal:success', data: number };
 
 // Publishes "cancel" to the "dismiss" topic
-Portals.publish(message);
+Portals.publish<Messages>({ topic: 'modal:dismiss', data: 'cancel' });
```

Comment:
Overall I really like the example. Should we also still show other valid uses like `Portals.publish({ topic: 'something:else', data: 'asdf' })`? I don't really have a huge opinion one way or the other.

Reply:
👍 FTFY